File: 2416-evaluate-boolean-binary-tree/evaluate-boolean-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def opp(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return True
        if root.left is None and root.right is None:
            if root.val==1:
                return True
            return False
        
        left= self.opp(root.left)
        right= self.opp(root.right)

        if root.val==2:
            return left or right
        
        return left and right

        
    def evaluateTree(self, root: Optional[TreeNode]) -> bool:
        return self.opp(root)

File: 2416-evaluate-boolean-binary-tree/evaluate-boolean-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean opp(TreeNode root){
        if(root.left==null && root.right==null){
            if(root.val==1)
                return true;
            return false;
        }

        boolean left = opp(root.left);
        boolean right = opp(root.right);

        if(root.val==2)
            return left || right;
        
        return left && right;
    }
    public boolean evaluateTree(TreeNode root) {
        return opp(root);
    }
}